modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java

```diff
@@ -2201,6 +2211,27 @@ public void testDuplicatedDiscoveryDataRemoved() throws Exception {
      */
     @Test
     public void testFailedNodeRestoreConnection() throws Exception {
+        assertFalse(checkRestoreConnection());
+
+        setLoggerDebugLevel();
+
+        assertTrue(checkRestoreConnection());
+    }
+
+    /** @throws Exception If failed.*/
+    private boolean checkRestoreConnection() throws Exception {
+        ListeningTestLogger testLog = new ListeningTestLogger(log);
+
+        LogListener lsnr = LogListener
+                .matches(Pattern.compile("Failed to ping node \\[.*\\]\\. Reached the timeout"))
+                .build();
+
+        testLog.registerListener(lsnr);
+
+        boolean result;
```

Comment:
Local variable is redundant. `return lsnr.check()` can be used. But it's better to pass additional parameter for `checkRestoreConnection` and check it at the end of the method, something like: `assertEquals(logExpected, lsnr.check())`